clang/lib/Sema/SemaInit.cpp

```diff
@@ -5634,6 +5634,9 @@ void InitializationSequence::InitializeFrom(Sema &S,
   bool allowObjCWritebackConversion = S.getLangOpts().ObjCAutoRefCount &&
          Entity.isParameterKind();
 
+  if (TryOCLSamplerInitialization(S, *this, DestType, Initializer))
```

Comment:
Sorry, could you please clarify: why do you need this change?

Reply:
To enable sampler initialization diagnostics in SYCL mode.

Reply:
Okay, I didn't see why it can't work on old place. I will expand hidden code next time.
Looks like we need also move `TryOCLZeroOpaqueTypeInitialization` if we want enable diagnostics for event in the future.

Reply:
Event type diagnostics should be enabled separately.